packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts

```diff
@@ -150,6 +150,19 @@ export const isSqlStrategy = (strategy: ModelDataSourceStrategy): strategy is SQ
   );
 };
 
+/**
+ * Provides the data source strategy for a given model
+ * @param ctx Transformer Context
+ * @param typename Model name
+ * @returns ModelDataSourceStrategyDbType
+ */
+export const getStrategyDbTypeFromModel = (ctx: DataSourceStrategiesProvider, typename: string): ModelDataSourceStrategyDbType => {
+  if (isBuiltInGraphqlType(typename)) {
+    return DDB_DB_TYPE;
```

Comment:
I am a little bit confused by the `DDB` type returned here. It seems that `isBuiltInGraphqlType` checks for the type `Query`, `Mutation` and `Subscription`, which are the types not annotated with `@model` and should not have an associated DB type. I can see this check is used in `isSqlModel` and `isDynamoDBModel` method which returns false. Is this just a default value returned here?

In addition, I think there is a validation in the transformer for the relational directives should be attached to the `@model` types (both parent and related). Are you aware of those validations when returning the default DB type for `Query` type here?

Reply:
Great question and thanks for calling this out!

Up front here's some context on the addition:
The relational transformers `graphql-has-many-transformer`, `graphql-has-one-transformer`, `graphql-belongs-to-transformer` were previously using [`isSqlModel`](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts#L136-L142) in a few places:
- [HasManyTransformer.prepare](https://github.com/aws-amplify/amplify-category-api/pull/2330/files#diff-04f1e3a6a8ce5bd950c9ebc3bc3ec562a443601d914dee53eeb82393825ba5d0L146)
- [HasOneTransformer.prepare](https://github.com/aws-amplify/amplify-category-api/pull/2330/files#diff-dd1c54a3c47c334e725948adce8ef86c20941fa2361ec6f864dc6c0c17f36a28L170)
- [setFieldMappingsReference](https://github.com/aws-amplify/amplify-category-api/pull/2330/files#diff-0110e5f5bea6c7251347e3013b183a69c2202d6756882aa750258296f0d617f2L247)

It was being used to determine whether the model is backed by a SQL or DDB data source. In refactoring this code, it was simpler to consistently work with a `ModelDataSourceStrategyDbType` rather than a `bool` with if / else branches. As you can see in existing usages of `isSqlModel` linked above, `true` was handled as SQL and `false` was handled as DDB. 

Based on your statements and questions, this logic seems incorrect. I presume that there's some validation happening further upstream preventing a non model type from being passed here.

Because this is a refactor, I'd like to keep this change as 100% behavioral parity with the existing implementation, even if that includes this incorrect logic. Then follow up with a fix. Is that reasonable or do you feel strongly that this should be addressed in this PR?

Reply:
Since it is the refactor, I agree the behavior parity consistence should be kept first. I am OK with it in current PR. The logic can be addressed separately